src/main/java/org/broadinstitute/hellbender/tools/spark/sv/discovery/inference/CpxVariantDetector.java

```diff
@@ -59,8 +60,17 @@ public void inferSvAndWriteVCF(final JavaRDD<AssemblyContigWithFineTunedAlignmen
         final JavaRDD<AnnotatedContig> annotatedContigs =
                 assemblyContigs.map(tig -> new AnnotatedContig(tig, referenceSequenceDictionaryBroadcast.getValue()));
 
-        SVVCFWriter.writeVCF(annotatedContigs.map(tig -> tig.toVariantContext(referenceBroadcast.getValue())).collect(),
-                outputPath, referenceSequenceDictionaryBroadcast.getValue(), toolLogger);
+        final List<VariantContext> complexVariants =
+                annotatedContigs
+                        .groupBy(tig -> new EquivKey(tig.getAffectedRefRegion(), tig.getAltSeq(), tig.getSegmentsAndDescription()))
+                        .map(pair -> {
+                            final VariantContext rawVariantContext = pair._1.toVariantContext(referenceBroadcast.getValue());
```

Comment:
I think it might be nice to extract this into a method called `produceCpxVariantContext(Tuple2<>...)`. Then it would be great to write a little unit test demonstrating the functionality (ie getting one variant context back with the right annotations set).